services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -547,6 +584,18 @@ async def _init_metadata() -> Tuple[int, str]:
         )
         return self.s3_client.create_presigned_put_url(bucket_name, object_name)
 
+    async def get_s3_link(self, user_id: str, file_uuid: str) -> str:
+        """
+        returns: the s3 file link and creates a databas entry for the file
+
+        NOTE: the user must call the update metadata endpoint to update
+        the metadata once the upload is finished
+        """
+        await self._generate_metadata_for_link(user_id=user_id, file_uuid=file_uuid)
+        bucket_name = self.simcore_bucket_name
+        object_name = file_uuid
+        return f"s3://{bucket_name}/{object_name.lstrip('/')}"
```

Comment:
what about using pydantic and AnyURL to ensure the link is valid?

Reply:
Agreed. We also found that ``pydantic.tools.parse_obj_as``  is very handy to validate against constraint types